elrond-wasm-modules/src/bonding_curve/utils/user_endpoints.rs

```diff
@@ -76,7 +74,7 @@ pub trait UserEndpointsModule: storage::StorageModule + events::EventsModule {
                     requested_amount.clone(),
                     bonding_curve.arguments.clone(),
                 );
-                let price_clone = price.clone()?;
+                let price_clone = price.clone();
```

Comment:
No need to clone price at all. Simply replace:
bonding_curve.payment_amount += price_clone;

With:
bonding_curve.payment_amount += &price;